Add additional information units

Jules-Bertholet opened the pull request:
The [shannon](https://en.wikipedia.org/wiki/Shannon_(unit)) is equivalent to the bit.
One [nat](https://en.wikipedia.org/wiki/Nat_(unit)) is log2(e) bits.
One [trit](https://en.wikipedia.org/wiki/Ternary_numeral_system) is 1 trinary digit (log2(3) bits).
One [hartley](https://en.wikipedia.org/wiki/Hartley_(unit)) is log2(10) bits.
The nonet (9 bits) is defined in [RFC 4042](https://datatracker.ietf.org/doc/html/rfc4042).
The declet (10 bits) is defined in [IEEE 754](https://ieeexplore.ieee.org/document/4610935).

hanna-kn commented:
Before adding a unit or a name/symbol, please ask yourself twice if the inclusion is really useful. Addition of a large number of units/names/symbols might harm usability.

- I'm particularly hesitant about all the *_digit units/names. 
- Composite units (not hidden) are primarily intended for easy access to common combinations of units and prefixes. While petabyte and exabyte are increasingly common units, extremely large binary prefixes are seldom used.
- I agree that "o" as abbreviation for byte/octet would be useful, particularly in a French context, although other uses might be desirable (as an alternative degree sign). "Sh" for shannon might however cause more confusion in comparison to its usefulness.

hanna-kn commented:
I suggest removal of the *_digit names and that "(* Digit)" is instead added to the titles (e.g. "Hartley (Decimal Digit)"). This way the user can still find the correct unit when searching. I doubt that anyone would actually wish type the full "trinary_digit" instead of "trit". Possibly "dit" could be added to the Hartley unit.

It might be more convenient to use tribit, triad, triade, or tribble as name for octal digit (although I'm not familiar with either).

Either "ao" (not presented to user) or "c" (case-sensitive) should be used for "Sh" (e.g. "c:Sh").

I think it would be reasonable to at least add petabyte, exabyte, and petabit again.

Jules-Bertholet commented:
I've added `dit` and made it the preferred form over `hartley` (to be consistent with `bit`). Also added petabyte, exabyte, petabit, and the -power-of-2 variants of these. I kept the `_digit` names because I feel they could still be useful for discovery using tab-completion in the terminal.
